Tests/SonarQube.Bootstrapper.Tests/BootstrapperExeTests.cs

```diff
@@ -45,6 +45,33 @@ public void Exe_ParsingFails()
         }
 
         [TestMethod]
+        public void Exe_PreProc_UrlIsRequired()
+        {
+            // Arrange
+            BootstrapperTestUtils.EnsureDefaultPropertiesFileDoesNotExist();
+            string rootDir = TestUtils.CreateTestSpecificFolder(this.TestContext);
+
+            using (InitializeNonTeamBuildEnvironment(rootDir))
+            {
+                string binDir = CalculateBinDir(rootDir);
+
+                MockBuildAgentUpdater mockUpdater = new MockBuildAgentUpdater();
+
+                // Act
+                TestLogger logger = CheckExecutionFails(mockUpdater, "begin");
+
+                // Assert
+                mockUpdater.AssertUpdateNotAttempted();
+                mockUpdater.AssertVersionNotChecked();
+
+                AssertDirectoryDoesNotExist(binDir);
+
+                logger.AssertErrorLogged(SonarQube.Bootstrapper.Resources.ERROR_Args_UrlRequired);
+                logger.AssertErrorsLogged(1);
+            }
+        }
```

Comment:
Higher-level test that checks the pre-processor fails if the url is not supplied.